### DISCUSSION TITLE
Q&A about CrowNet

### DISCUSSION THREAD

**Discussion by stefanSchuhbaeck** [roVer-HM/crownet#12] [Q&A] [answered]
If there are questions please just comment here or open up a new discussion.

**Comment by gkisaisik** [comment]
Hello,

Thank you for sharing this project. I believe pedestrian simulation in OMNeT++ has been a challenging area, and your work will significantly help researchers.

In the light of resources, I have set up the environment, yet encountered  certain isues given below. Please find my progress in steps below:
1. In the root directory, I created the workspace directory (**omnetpp-ws**).
2. I imported the necessary subprojects: **inet4, crownet, simu5g, artery**, and **veins**. (but when importing the veins folder, I couldn’t find modules 1 (veins) and 3 (inet) specifically, as they were not visible within the veins file.)
3. In different terminals, I successfully ran **omnetpp-ide, flowcontrol-ide** (I noticed that you advised running **control-ide**, but I couldn’t find it), and **sumo-ide**. 

When I attempted to run **omnetpp.ini** in the **testSim** directory, the mentioned errors as follows:
- **VeinsInetManager** module is not found ("No such module type").
- NED Consistency Problems: Multiple errors related to missing or unresolved NED types, specifically in the **Highway.ned** file, as well as others like **artery_core, artery_envmod, artery_storyboard**, and **traci**.

Please find a screenshot of the errors displayed in the IDE for your reference below. It seems that there might be missing dependencies or modules that are not correctly linked during the import process.

I would greatly appreciate it if you could help me in resolving these errors.

Best regards.
Gizay

![hata1](https://github.com/user-attachments/assets/d67800db-d75a-4a4a-81cb-2c9b7e43903f)

**Comment by stefanSchuhbaeck** [comment, marked as answer]
The problem is that you have to add the nested projects. For that tick the box **Search for nested projects**. See the image below. 
![image](https://github.com/user-attachments/assets/cb924cbb-70f2-4cbe-90ad-235735251e9e)


> In different terminals, I successfully ran omnetpp-ide, flowcontrol-ide 
> (I noticed that you advised running control-ide, but I couldn’t find it), and sumo-ide.

* We do not provide a  sumo-ide container as we normally do not change anything in the Sumo source code.  Therefore we only provide an executable wrapped in a container. 
* `control-ide` this must be a typo. We only have the `flowcontrol` container I will fix that.  Thanks for finding it. 


> VeinsInetManager module is not found ("No such module type").

This will be fixed when you include the submodule  from veins. 

> NED Consistency Problems:  ... Highway.ned
This will be fixed when you include the submodule  from veins. Look into the path `org.car2x.veins.subprojects...`

> as well as others like artery_core, artery_envmod, artery_storyboard, and traci.

* Go into the `artery` directory and check that there are files and directories.  
* Do the same for `artery/extern/vaneza`

If these directories are empty ensure that the git submodules are working.  In the `artery` directory 

```
$ git submodule status
 8de7772cc72daca8e947b79b83fea46214931604 extern/pybind11 (v2.6.2)
-7713e34d0b1f1c49e533bd4c5007082ab6fa7e4d extern/sim0mqpp
 4fe66ece854c8ba7f5ce6ff20167327fb780d4e6 extern/vanetza (heads/hm_master)
 dd59d015c57ca02fc6f984a7b2ec0ebb3be1240c scenarios/mt-its2017 (heads/master)
```

It should look like this (most likely different hashes but four lines). If not  initialize the submodules 

```
git submodule init 
git submodule upate
```

For overall OMNeT  take a look at the [documentation](https://omnetpp.org/documentation/) provided by OMNeT directly. For the IDE see this [PDF](https://doc.omnetpp.org/omnetpp/UserGuide.pdf).

**Comment by gkisaisik** [comment]
Thank you for the detailed response. The issue has now been resolved.
